src/api/metadata.coffee

```diff
@@ -0,0 +1,161 @@
+Channel = require('../model/channels').Channel
+Client = require('../model/clients').Client
+Mediator = require('../model/mediators').Mediator
+User = require('../model/users').User
+ContactGroup = require('../model/contactGroups').ContactGroup
+
+Q = require 'q'
+logger = require 'winston'
+authorisation = require './authorisation'
+utils = require "../utils"
+
+# Map string parameters to collections
+collections =
+  Channels: Channel
+  Clients: Client
+  Mediators: Mediator
+  Users: User
+  ContactGroups: ContactGroup
+
+
+#Function to remove properties from export object
+removeProperties = (obj) ->
+  propertyID = '_id'
+  propertyV = '__v'
+  
+  for prop of obj
+    if (prop == propertyID || prop == propertyV)
+      delete obj[prop]
+    else if ( typeof obj[prop] == 'object' || obj[prop] instanceof Array )
+      removeProperties(obj[prop])
+  return obj
+
+
+# Function to return unique identifier key and value for a collection
+getUniqueIdentifierForCollection = (collection, doc) ->
+  switch collection
+    when 'Channels' then uidKey = 'name'; uid = doc.name
+    when 'Clients' then uidKey = 'clientID'; uid = doc.clientID
+    when 'Mediators' then uidKey = 'urn'; uid = doc.urn
+    when 'Users' then uidKey = 'email'; uid = doc.email
+    when 'ContactGroups' then uidKey = 'groups'; uid = doc.groups
+  returnObj = {}
+  returnObj[uidKey] = uid
+  return returnObj
+
+
+# Build response object
+buildResponseObject = (model, doc, status, message, uid) ->
+  return {
+    model: model
+    record: doc
+    status: status
+    message: message
+    uid: uid
+  }
+
+# API endpoint that returns metadata for export
+exports.getMetadata = () ->
+  # Test if the user is authorised
+  if not authorisation.inGroup 'admin', this.authenticated
+    utils.logAndSetResponse this, 403, "User #{this.authenticated.email} is not an admin, API access to getUsers denied.", 'info'
+    return
+
+  try
+    exportObject = {}
+    params = this.request.query
+    
+    # Return all documents from all collections for export
+    for col of collections
+      exportObject[col] = yield collections[col].find().lean().exec()
+      for doc of exportObject[col]
```

Comment:
I think you are meaning to use `for ... in array` here and line 69 instead of `for ... of`. After that line 72 and 73 could probably be `if doc._id`. Unless I'm missing something.